core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -310,9 +316,12 @@ public synchronized IndexShard createShard(ShardRouting routing) throws IOExcept
             eventListener.indexShardStateChanged(indexShard, null, indexShard.state(), "shard created");
             eventListener.afterIndexShardCreated(indexShard);
             indexShard.updateRoutingEntry(routing, true);
-            if (shards.isEmpty() && this.indexSettings.getTranslogSyncInterval().millis() != 0) {
+            if (shards.isEmpty()) {
                 ThreadPool threadPool = nodeServicesProvider.getThreadPool();
-                new AsyncTranslogFSync(this, threadPool).schedule(); // kick this off if we are the first shard in this service.
+                if (this.indexSettings.getTranslogSyncInterval().millis() != 0) {
+                    new AsyncTranslogFSync(this, threadPool); // kick this off if we are the first shard in this service.
+                }
+                rescheduleRefreshTasks();
```

Comment:
Hmm why do we need to do this?  The already scheduled one from the ctor should be the same?  Only if the refresh interval is changed should we need to reschedule?